### ISSUE TITLE
object-curly-spacing with line comment

### ISSUE THREAD

**Issue by alfredking1988** [eslint/eslint#11902]
**Tell us about your environment**

* **ESLint Version:** 6.0.1
* **Node Version:** 10.16.0
* **npm Version:** 6.8.0

**What parser (default, Babel-ESLint, etc.) are you using?**
default

**Please show your full configuration:**

<details>
<summary>Configuration</summary>

```js

{
  root: true,
  rules: {
    'object-curly-spacing': 'error',
  },
}

```

</details>

**What did you do? Please include the actual source code causing the issue, as well as the command that you used to run ESLint.**

```js
var o = { // hello, this is the line comment, which should be ok
  n: 42,
};
```

```bash
eslint test.js
```

**What did you expect to happen?**
ESLint should pass without error

**What actually happened? Please include the actual, raw output from ESLint.**
ESLint output error: 
`1:9  error  There should be no space after '{'  object-curly-spacing`

**Note**
Maybe there should an option for this rule, to allow space after the `{` if trailing comment exists?

**Comment by g-plane** [comment]
I can reproduce this bug: [Reproduction Link](https://eslint.org/demo/#eyJ0ZXh0IjoidmFyIG8gPSB7IC8vIGhlbGxvLCB0aGlzIGlzIHRoZSBsaW5lIGNvbW1lbnQsIHdoaWNoIHNob3VsZCBiZSBva1xyXG4gIG46IDQyLFxyXG59OyIsIm9wdGlvbnMiOnsicGFyc2VyT3B0aW9ucyI6eyJlY21hVmVyc2lvbiI6NSwic291cmNlVHlwZSI6InNjcmlwdCIsImVjbWFGZWF0dXJlcyI6e319LCJydWxlcyI6eyJvYmplY3QtY3VybHktc3BhY2luZyI6Mn0sImVudiI6e319fQ==).

**Comment by kaicataldo** [comment]
It does seem to me like the rule should ignore white space around comments after the opening brace and before the closing brace when there are line breaks.

**Comment by mdjermanovic** [comment]
> It does seem to me like the rule should ignore white space around comments after the opening brace and before the closing brace when there are line breaks.

Ignore the brace regardless of the options if the first non-comment token is on another line?

**Comment by silverwind** [comment]
This is a regression in 6.x, it did pass in 5.x. I suspect https://github.com/eslint/eslint/commit/67c08b67509c54acd96aab2cec22efb53bfe6265 introducted the regression.

CC: @g-plane

**Comment by mdjermanovic** [comment]
I don't think this is a bug, the rule now works with comments same as with all other tokens.

Nevertheless, there is already a precedent for this particular exception, e.g. `space-in-parens` allows this:

```js
/*eslint space-in-parens: ["error", "never"]*/

var foo = ( // No errors. The option is 'never', but spaces between '(' and '//' are allowed
  bar
);
```

The above is the only exception, the following are still errors:

```js
/*eslint space-in-parens: ["error", "always"]*/
var foo = (//
  bar
);

/*eslint space-in-parens: ["error", "always"]*/
var foo = (/**/ 
  bar
);

/*eslint space-in-parens: ["error", "always"]*/
var foo = (
  bar
/**/);

/*eslint space-in-parens: ["error", "never"]*/
var foo = ( /**/
  bar
);

/*eslint space-in-parens: ["error", "never"]*/
var foo = ( 
  bar
/**/ );
```

I think it's okay to allow only this particular exception (`never` + line comment) in `object-curly-spacing` as a small change in the default behavior.

**Comment by mdjermanovic** [comment]
`block-spacing` and `comma-spacing` also have exception for line comments. 

I'm working on this.
